### PR TITLE
Intercom secret project

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Preparation for intercom  secret project.
  - [x] Connection and volume persistance on a seat level.
  - [x] Minor bugfixes (for 2.5)
  - [ ] classified 1
  - [ ] classified 2

### CODE REVIEW

**Comment by Cuel** [comment]
🤔 classified

File: addons/sys_intercom/fnc_configIntercomStatus.sqf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Author: ACRE2Team
+ * Configures the initial intercom connectivity (not connected, connected to crew intercom, connected to pax intercom or connected to both) for all allowed seats. *
+ *
+ * Arguments:
+ * 0: Vehicle with intercom <OBJECT>
+ * 1: Positions with intercom <ARRAY>
+ * 2: Intercom <STRING><OPTIONAL>
```
Comment (jonpas) [inline addons/sys_intercom/fnc_configIntercomStatus.sqf]: `<STRING> (default: value)`
This only uses first one though?

File: addons/sys_intercom/fnc_configRxTxCapabilities.sqf [hunk 2]
```diff
@@ -1,16 +1,15 @@
 /*
  * Author: ACRE2Team
- * Returns the configuration (No use, TX Ony, RX only or TX/RX) for the given unit of a radio that is connected to an intercom.
+ * Configures a the initial state (No use, TX Ony, RX only or TX/RX) of a rack that is connected to intercom.
```
Comment (jonpas) [inline addons/sys_intercom/fnc_configRxTxCapabilities.sqf]: `a the`, `Ony`?

File: addons/sys_intercom/fnc_getIntercomConnectionStatus.sqf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * Author: ACRE2Team
+ * Gets the intercom connection status of the vehicle seat the unit is in.
+ *
+ * Arguments:
+ * 0: Vehicle with intercom <OBJECT>
+ * 1: Unit to be checked <OBJECT>
+ *
+ * Return Value:
+ * Intercom connection status <NUMBER>
+ *  0: no intercom
```
Comment (jonpas) [inline addons/sys_intercom/fnc_getIntercomConnectionStatus.sqf]: Style with `- ` maybe, this looks like it's an array (yes I know it says `NUMBER`).

File: addons/sys_intercom/fnc_intercomAction.sqf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * Adds an action for joining the passenger intercom.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorTarget] call acre_sys_intercom_iIntercomAction
```
Comment (jonpas) [inline addons/sys_intercom/fnc_intercomAction.sqf]: `iIntercom`

File: addons/sys_intercom/fnc_intercomAction.sqf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_target"];
+
+private _type = typeOf _target;
+
+// Exit if object has no intercom
+if (getNumber (configFile >> "CfgVehicles" >> _type >> "acre_hasPassengerIntercom") != 1 && getNumber (configFile >> "CfgVehicles" >> _type >> "acre_hasCrewIntercom") != 1) exitWith {};
```
Comment (jonpas) [inline addons/sys_intercom/fnc_intercomAction.sqf]: Lazy eval on second part?

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**